src/test/java/io/kabanero/KabaneroToolManagerTest.java

```diff
@@ -0,0 +1,78 @@
+package io.kabanero;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.UUID;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import io.kubernetes.KabaneroClient;
+import io.kubernetes.client.ApiException;
+
+import static org.junit.Assert.*;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(KabaneroClient.class)
+public class KabaneroToolManagerTest {
+    KabaneroInstance kabaneroInstance;
+    KabaneroToolManager kManTool;
+
+    @Before
+    public void mockKabaneroClient() throws IOException, ApiException, GeneralSecurityException {
+        PowerMockito.mockStatic(KabaneroClient.class);
+        Mockito.when(KabaneroClient.getInstances()).thenReturn(new ArrayList<KabaneroInstance>());
```

Comment:
if you look in `KabaneroToolManager. getKabaneroToolManagerInstance()` it actually calls `KabaneroClient.discoverTools(SINGLE_TOOL_MANAGER_INSTANCE);` which is the method you should be mocking, instead of `KabaneroClient.getInstances()` which the KabaneroToolManager doesn't call.